### PR TITLE
fixing package issues and adding SplitIntoSubnetsIPv4 unit tests

### PR DESCRIPTION
**Pull request by luthermonson**
**What type of PR is this?**
Fixing the cidr package a bit, the tests had a weird package/import which made no sense and since I was in the `SplitIntoSubnetsIPv4` code anyway trying to figure out what it does I wrote some unit tests for it.

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: pkg/internal/cidr/cidr_test.go [hunk 1]
```diff
@@ -14,16 +14,86 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cidr_test
+package cidr
```
Comment (Ankitasw) [inline pkg/internal/cidr/cidr_test.go:17]: It's better we keep this pkg name as before.

**Review by Skarlso — COMMENTED** [review]
Ah I submitted this comment last night but it was pending 🤦

**Comment by Skarlso** [comment]
/hold
until tests

**Comment by luthermonson** [comment]
Why did we approve it but argue about me changing the package name? Am I changing it back or not?

**Comment by Skarlso** [comment]
> Why did we approve it but argue about me changing the package name? Am I changing it back or not?

People can have different opinions, right? :D

I, for one, don't really care about it as long as the tests are passing and the coverage didn't suffer.

If Ankita feels strongly about this though, then I'm with her. :)

**Comment by luthermonson** [comment]
@Ankitasw your call, I personally don't think there is any benefit in this package to do an "external test" in the same directory.

**Comment by Ankitasw** [comment]
> your call, I personally don't think there is any benefit in this package to do an "external test" in the same directory.
 
It is followed the same way in other packages as well, so I wanted to be consistent about it. Otherwise we have to change that in other packages as well.

**Comment by Ankitasw** [comment]
> Why did we approve it but argue about me changing the package name? Am I changing it back or not?

I agree with @Skarlso, everybody from community can give feedback 😄

**Comment by pydctw** [comment]
Interesting conversation. I've done both but considering 
* We've been using same package names for test files in CAPA - [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/pkg/cloud/services/ec2), [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/pkg/internal)
* CAPI also uses same package names for test files - [here](https://github.com/kubernetes-sigs/cluster-api/tree/main/util/)

I think we should keep @luthermonson's change of using `cidr` as a package name for the test file.

**Comment by pydctw** [comment]
There is a linter error. Could you fix it?

**Comment by Ankitasw** [comment]
> We've been using same package names for test files in CAPA - [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/pkg/cloud/services/ec2), [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/pkg/internal)
CAPI also uses same package names for test files - [here](https://github.com/kubernetes-sigs/cluster-api/tree/main/util/)

I think only for ec2 service we have made exception, I didn't have strong opinion @luthermonson , it was just a convention I would like to follow.

**Comment by luthermonson** [comment]
@pydctw yup! thanks ill fix

**Comment by luthermonson** [comment]
@pydctw gtg

**Comment by pydctw** [comment]
All tests passed
/hold cancel

**Pull request merged**